### PR TITLE
Update build workflow with cross build

### PR DESCRIPTION
**Pull request by chandrareddyp**
### What this PR does / why we need it
PR (Pull Request) is created for the following changes:

1. Updated the Main workflow with `make cross-build` and `make lint`, which were moved from the "Tanzu CLI Unit Tests" workflow.
2. Updated the name of the Main workflow to "Build" from "Main."
3. Removed `make gomod` from the "Tanzu CLI Unit Tests" workflow, as it is already covered in the `make verify` step in the Main or Build workflow.

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR

- All the PR workflows/Checks are executed successfully

### Release note

```release-note

```

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -56,3 +56,7 @@ jobs:
     - name: Run make verify
       run: |
         make verify
+
+    - name: Cross build
+      run: |
+        make cross-build
```
Comment (anujc25) [inline .github/workflows/build.yaml:62]: nit: Missing new line at the end

**Review by anujc25 — COMMENTED** [review]
Thanks for updating this. LGTM 
(just a minor comment regarding failing CI)

**Review by mpanchajanya — COMMENTED** [review]
Wondering if we should add `make lint` and `make build` steps in `build .yaml` after `make verify` ?

**Comment by chandrareddyp** [comment]
> 

Updated. thanks.

**Comment by vuil** [comment]
Are the latest changed in sync with the PR description. I only see a single line change in build.yaml

**Comment by chandrareddyp** [comment]
> Are the latest changed in sync with the PR description. I only see a single line change in build.yaml

@vuil the build was failing, so this morning I reverted changes for testing purpose. I have updated the PR with actual changes.

**Pull request merged**